ceph/ceph_admin/helper.py

```diff
@@ -391,6 +393,117 @@ def generate_rgw_spec(self, spec):
 
         return template.render(spec=spec)
 
+    def exec_command(self, node, **kwargs: Any) -> Tuple:
+        """
+        Return the results of the command executed.
+
+        Args:
+            node:   The node to be used for executing the given command.
+            kwargs: The supported keys are
+                    sudo    - Default false, root privilege to be used or not
+                    cmd     - The command to be executed
+                    check_ec - Evaluate the error code returned.
+
+        Return:
+            out, err    String values
+
+        Raises:
+            CommandFailed   when there is an execution failure
+        """
+        out, err = node.exec_command(**kwargs)
+
+        out = out.read().decode() if isinstance(out, ChannelFile) else out
+        err = err.read().decode() if isinstance(err, ChannelFile) else err
+        return out, err
+
+    def generate_snmp_spec(self, spec):
+        """
+        Return spec content for snmp-destination
+
+        Args:
+            spec (Dict): snmp-destination service spec config
+
+        Returns:
+            service_spec (Str)
+
+        Example::
+
+        specs:
+              - service_type: snmp-destination
+                spec:
+                  credentials:
+                    snmp_v3_auth_username: myadmin
+                    snmp_v3_auth_password: mypassword
+                    engine_id: fsid
+        """
+        template = self._get_template("snmp")
+        destination_node = spec["spec"].pop("snmp_destination", None)
+        node = get_node_by_id(self.cluster, destination_node)
+        if destination_node:
+            spec["spec"]["snmp_destination"] = self.get_addr(node) + ":162"
```

Comment:
would it be good to pass custom snmp port number( 161 | 162 ) rather than having static code?

Reply:
No @sunilkumarn417 . It is mentioned to have port 162 in all documentation for testing SNMP support for RHCS